samples/subsys/usb/dfu/README.rst

```diff
@@ -0,0 +1,110 @@
+.. _usb_dfu:
+
+USB DFU Sample Application
+##########################
+
+Overview
+********
+
+This sample app demonstrates use of a USB DFU Class driver provided
+by the Zephyr project.
+
+Requirements
+************
+
+This project requires an USB device driver. Currently, the USB DFU
+class provided by the Zephyr project depends on DFU image manager and
+partition layout. Refer to :ref:`flash_partitions` for details about
+partition layout. You SoC must run MCUboot as stage 1 bootloader.
+This sample is build as an application for the MCUboot bootloader.
+
+Building and Testing
+********************
+
+Building and signing the application
+====================================
+
+This sample can be built in the usual way (see :ref:`build_an_application`
+for more details) and flashed with regular flash tools, but will need
+to be loaded at the offset of SLOT-0.
+
+Application images (such as this sample) must be signed.
+Use the ``scripts/imagetool.py`` script from the `MCUboot GitHub repo`_
+to sign the image.  (See the `Using MCUboot with Zephyr`_ documentation for
+details.)
+
+.. code-block:: console
+
+   ~/src/mcuboot/scripts/imgtool.py sign \
+           --key ~/src/mcuboot/root-rsa-2048.pem \
+           --header-size 0x200 \
+           --align 8 \
+           --version 1.2 \
+           --included-header \
+           ./zephyr/zephyr.bin \
+           signed-zephyr.bin
+
+Build and flash MCUboot bootloader for Zephyr project as it is described in
+the `Using MCUboot with Zephyr`_ documentation. Then build, sign and flash
+the USB DFU sample at the offset of SLOT-0.
+
+Build and sign a second application image e.g. :ref:`hello_world`,
+which will be used as an image for the update.
+Please do no forget to enable the required MCUboot Kconfig option as described
```

Comment:
Don't need to say "please".  Add a blank line to start a new paragraph and say,

    Enable the MCUboot Kconfig option (as described in :ref:`mcuboot`) by
    adding the following line to your local copy of :file:`samples/hello_world/prj.conf`:

Actually, isn't there a way to do this with a project conf file rather than making
changes to the sample's prj.conf file?

Reply:
prj.conf should be changed before a second application is built, I think there should not be a blank line in between.

Reply:
@dbkinder "Please" has been removed and I think @jfischer-phytec-iot 's point is valid.